### PR TITLE
update deps

### PR DESCRIPTION
**Pull request by gabriel-v**
Scylla v0.14 and all of the others

Added new dev dependency for running all the tests that were starting with "truncate table" serially instead of in parallel ( i got some errors until i figured out to run with --test-thread=1 )

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,21 +19,22 @@ categories = ["database"]
...
-heck = "0.4"
-tokio = { version = "1", features = ["time", "rt-multi-thread", "io-util"] }
-thiserror = "1"
-futures-util = "0.3"
-syn = { version = "2", features = ["full", "extra-traits"] }
-quote = "1"
-proc-macro2 = "1"
-serde = { version = "1", features = ["derive"] }
-serde_json = "1"
+scylla = "0.14.0" # This crate is specifically build for this version
```
Comment (Jasperav) [inline Cargo.toml:22]: This specific version dependency is fine, for the others, please remove the patch version :) so x.y and not x.y.z

**Review by Jasperav — CHANGES_REQUESTED** [review]
Thanks for the PR! If you made the changes to the dependency and bump the Cargo.toml, I will publish the new version.

**Comment by gabriel-v** [comment]
ok

bumped workspace version 0.2.0, there are breaking changes:
- `select_paged` has different argument type
- the "Cursor" pub typedef was not used anymore

**Comment by Jasperav** [comment]
@gabriel-v LGTM, do you know why this error occurs on the CI? https://github.com/Jasperav/Catalytic/actions/runs/11437836428/job/31829718889?pr=48. If the CI is ok, I will publish the new version!

**Comment by gabriel-v** [comment]
here you go

https://github.com/gabriel-v/Catalytic/actions/runs/11445259484

errors in order:
- cargo lock version = 4 was from my local nightly compiler, when switched to stable, it now outputs version = 3
- muted some clippy errors about comment string formatting, unrelated to patch
- muted clippy error on dead code (Field::attribute, undocumented, for macro?) - plz review
- removed "... and type is not null" from materialized view example - this errors (as type is not PK) in scylla >5.4.X, 6.x but not scylla 5.2.X nor cassandra 3.11
- updated CI compiler toolchain to the newer one i had locally, since the compiler error text tests depend closely on compiler version (different structs implementing missing trait get printed)

**Comment by Jasperav** [comment]
Thanks a lot @gabriel-v, I published the changes!

**Pull request merged**